proxmox_kvm: 500 error args parameter

almdudler777 opened the pull request:
##### SUMMARY
vm_args should not be added as 'args' if proxmox_default_behavior is set to no_defaults.

Otherwise the API might always throw 500 - Error because only root@pam is permitted to set 'args'.
This is still tricky, as the Proxmox API does not allow API Tokens to be used when setting "root-only" parameters, even if the user of the token is 'root@pam'. 

Will fix this in a separate issue, as multiple  other parameter are affected by this.

Also we should never add 'force' parameter, even if set to false, the Proxmox API seems to expect the 'archive' parameter if 'force' is submitted. The 'archive' parameter is not implemented at all and will result in a '400 Bad Request: Parameter verification failed'.
In the Proxmox API 'force' is always assumed to be 'false | 0' anyways.

Fixes #1641 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
proxmox_kvm

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
1. Problem when always supplying force:
```
   - name: "Create VM"
      ignore_errors: yes
      community.general.proxmox_kvm: 
        api_user: root@pam
        api_token_id: "ansible"
        api_token_secret: "38***********0-8e8b-ef63135bbe4d"
        api_host: "****"
        state: present
        validate_certs: no
        node: "******"
        vmid: 9000
      register: exec_result

"msg": "creation of qemu VM None with vmid 9000 failed with exception=400 Bad Request: Parameter verification failed. - b'{\"errors\":{\"archive\":\"missing property - \\'force\\' requires this property\"},\"data\":null}'"
```

2. providing proxmox_default_behavior will prevent 'force' from being added, but results in 500 Error because I'm using API Token for root@pam which is not equivalent to using root@pam with password....
```
   - name: "Create VM"
      ignore_errors: yes
      community.general.proxmox_kvm: 
        api_user: root@pam
        api_token_id: "ansible"
        api_token_secret: "38***********0-8e8b-ef63135bbe4d"
        api_host: "****"
        state: present
        validate_certs: no
        node: "******"
        vmid: 9000
        proxmox_default_behavior: "no_defaults"
      register: exec_result

"msg": "creation of qemu VM None with vmid 9000 failed with exception=500 Internal Server Error: only root can set 'args' config - b'{\"data\":null}'"
```

3. with my fixes, the vm will be created as expected
```
   - name: "Create VM"
      ignore_errors: yes
      community.general.proxmox_kvm: 
        api_user: root@pam
        api_token_id: "ansible"
        api_token_secret: "38***********0-8e8b-ef63135bbe4d"
        api_host: "****"
        state: present
        validate_certs: no
        node: "******"
        vmid: 9000
        proxmox_default_behavior: "no_defaults"
      register: exec_result

"msg": "VM None with vmid 9000 deployed",
```

almdudler777 commented:
Uff, sorry for the mess and the rough start :) 

Still new to git and had wrong / non-github email setup for git on my ansible machine - so the commits were not under the right account.

Not sure how to fix the "needs_rebase"?

Akasurde commented:
@almdudler777 You can follow this guide for rebasing your pull request - https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html.

almdudler777 commented:
@Akasurde thanks so much for the help! I hope i did it correctly now. man, git is a beast :) i suddenly had every commit 3 times in my git log. I rebased from 562ff7efb7e6c69bfaa4c534693c70425cdc1e6c upwards and dropped everything except the three commits actually carrying the correct changes + correct account info.

One question though, would you rather have everything in one commit? I saw a squash option while rebaseing. Or is it ok to have multiple commits in one PR? (one project likes it that way others don't really care...) just asking because i plan on doing some more stuff on this module...

felixfontein commented:
@almdudler777 multiple commits are fine, we squash on merging.

felixfontein reviewed (commented):
Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)?

felixfontein commented on changelogs/fragments/1783-proxmox-kvm-fix-args-500-error.yaml:
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - proxmox_kvm - do not add ``args`` if ``proxmox_default_behavior`` is set to no_defaults  (https://github.com/ansible-collections/community.general/issues/1641).
+  - proxmox_kvm - never implicitly add ``force`` equal to False, Proxmox API requires not implemented parameters otherwise, and assumes force to be false by default anyways. 
```
```suggestion
  - proxmox_kvm - stop implicitly adding ``force`` equal to ``false``. Proxmox API requires not implemented parameters otherwise, and assumes ``force`` to be ``false`` by default anyways (https://github.com/ansible-collections/community.general/pull/1783).
```

felixfontein commented:
@Aversiste can you please take a look?

Aversiste reviewed (approved):
I am ok with this change.

felixfontein commented:
@almdudler777 thanks for fixing this!
@Akasurde @Aversiste thanks for reviewing!
